### ISSUE TITLE
Confluent Schema Registry as a schema uri

### ISSUE THREAD

**Issue by rbolkey** [logstash-plugins/logstash-codec-avro#14]
We keep our Avro schemas in Confluent's [Schema Registry](http://docs.confluent.io/2.0.0/schema-registry/docs/index.html). It would be great if we could point the `schema_uri` to the registry's API, but the registry returns the Avro schema as a nested JSON object with the key `schema`:

```
HTTP/1.1 200 OK
Content-Type: application/vnd.schemaregistry.v1+json

{
  "schema": "{\"type\": \"string\"}"
}
```

Thoughts on what would be the best way to support this?

**Comment by malonej7** [comment]
I agree, this is needed for us as well.  It looks like there's a new codec that lets you point to the Schema Registry in the logstash Input (https://github.com/revpoint/logstash-codec-avro_schema_registry).  But we need a way in the logstash Output to send to Kafka using avro with a schema_uri pointing to our confluent schema registry.  Is this possible, or is anyone working on this already?

**Comment by talevy** [comment]
@suyograo shouldn't this be implemented in revpoint's repository? https://github.com/revpoint/logstash-codec-avro_schema_registry/blob/master/lib/logstash/codecs/avro_schema_registry.rb#L90

**Comment by tomsej** [comment]
+1 Confluent platform seems to becoming popular and revpoint repository is unfortunately outdated. Do you consider to create your own avro codec with schema registry support?

There are two problems:

1) [Schema Registry](http://docs.confluent.io/3.0.0/schema-registry/docs/index.html) API returns the Avro schema in `schema` object. You expect that the Avro schema is not inside of nested object.

2) Every message inside of Confluent Platform should have following structure:
`< magic byte > < schema id (4 bytes) > < Avro blob >`
Currently the codec is not prefixing the Avro blob with magic byte nor schema id.

It would be really great to have this codec available. Unfortunately I know nothing about Ruby :/

**Comment by bjornhjelle** [comment]
+1

**Comment by accbel** [comment]
+1

**Comment by CBR09** [comment]
Anyone working on this?,  same issue like @malonej7, could you solve or workaround that?

**Comment by ryananguiano** [comment]
@CBR09 just took a first crack at it.

https://github.com/revpoint/logstash-codec-avro_schema_registry/pull/5

**Comment by danielyahn** [comment]
Any update on supporting AVRO encoding when outputting to Kafka?

**Comment by shah-smit** [comment]
Hi Team, is there any update or making the [this](https://github.com/revpoint/logstash-codec-avro_schema_registry) official?

**Comment by syepes** [comment]
@andsel Is there any news or final decision on this support?
The Confluent's [Schema Registry](http://docs.confluent.io/2.0.0/schema-registry/docs/index.html) has now become kind of a standard when using Kafka

**Comment by andsel** [comment]
No, however you can use the Confluent Schema Registry directly from Kafka input/output plugins.
